File: web/src/api/task/task.js
import service from '@/utils/request'

export const listTask = (data, subType) => {
  return service({
    url: '/task/list?type=' + subType,
    method: 'post',
    data
  })
}

export const listReviewTask = (data, subType) => {
  return service({
    url: '/task/review?type=' + subType,
    method: 'post',
    data
  })
}

export const listHistoryTask = (data, subType) => {
  return service({
    url: '/task/history?type=' + subType,
    method: 'post',
    data
  })
}
export const getTask = (data, subType) => {
  return service({
    url: '/task?id=' + data + '&?type=' + subType,
    method: 'get',
  })
}

export const updateTask = (data) => {
  return service({
    url: '/task',
    method: 'put',
    data
  })
}

export const createTask = (data) => {
  return service({
    url: '/task',
    method: 'post',
    data
  })
}

File: go/router/routers/task.go
package routers

import (
	"github.com/gin-gonic/gin"
	v1 "github.com/qingfeng777/owls/server/api/v1"
	"github.com/qingfeng777/owls/server/middleware"
)

func (s *Routers) InitTaskRouter(Router *gin.RouterGroup) {
	recordRouter := Router.Use(middleware.OperationRecord())

	{
		taskApi := v1.ApiGroupApp.Task

		recordRouter.POST("/task", taskApi.AddTask)
		recordRouter.PUT("/task", taskApi.UpdateTask)
		Router.GET("/task", taskApi.GetTask)
		Router.POST("/task/list", taskApi.ListTask)
		Router.POST("/task/review", taskApi.ListReviewTask)
		Router.POST("/task/history", taskApi.ListHistoryTask)
	}
}
